11 login register page form

KacperWojdak opened the pull request:
Form for login and register is created and ready
![2](https://github.com/user-attachments/assets/776040ae-aee0-404c-9e17-92fd684430be)
![1](https://github.com/user-attachments/assets/9d4e6c01-6278-4714-890b-f411f01042ea)

KacperWojdak committed:
Created Authentication Form

KacperWojdak committed:
Code cleaning

PiotrFedak commented on frontend/src/pages/LoginPage.js:
```diff
@@ -1,16 +1,90 @@
+    setIsSignUp(!isSignUp)
+  }
+
+  const loginForm = (
+    <form className={`w-full max-w-md mx-auto bg-white p-8 rounded-lg shadow-md transition-opacity duration-500 ease-in-out ${isSignUp ? 'opacity-0 pointer-events-none' : 'opacity-100'}`}> 
+      <h3 className="text-2xl font-bold text-center mb-6">Logowanie</h3>
+      <input
+        className="w-full p-4 mb-4 text-gray-700 border rounded-md"
+        type="email"
+        placeholder="E-mail"
+        required
+      />
+      <input
+        className="w-full p-4 mb-4 text-gray-700 border rounded-md"
+        type="password"
+        placeholder="Password"
+        required
+      />
+      <button
+        type="submit"
+        className="w-full p-4 bg-blue-500 text-white rounded-md hover:bg-blue-600 transition"
+      >
+        Zaloguj się
+      </button>
+      <p className="text-center mt-4">
+        Nie posiadasz konta?{' '}
+        <a href="#" onClick={toggleForm} className="text-blue-500 underline">
+          Zarejestruj się.
+        </a>
+      </p>
+    </form>
+  )
```
This should be in seperate file in components/Login

KacperWojdak replied:
✔️

PiotrFedak commented on frontend/src/pages/LoginPage.js:
```diff
@@ -1,16 +1,90 @@
+    </form>
+  )
+
+  const registerForm = (
+    <form className={`w-full max-w-md mx-auto bg-white p-8 rounded-lg shadow-md transition-opacity duration-500 ease-in-out ${isSignUp ? 'opacity-100' : 'opacity-0 pointer-events-none'}`}> 
+      <h3 className="text-2xl font-bold text-center mb-6">Rejestracja</h3>
+      <input
+        className="w-full p-4 mb-4 text-gray-700 border rounded-md"
+        type="text"
+        placeholder="Username"
+        required
+      />
+      <input
+        className="w-full p-4 mb-4 text-gray-700 border rounded-md"
+        type="email"
+        placeholder="E-mail Address"
+        required
+      />
+      <input
+        className="w-full p-4 mb-4 text-gray-700 border rounded-md"
+        type="password"
+        placeholder="Create Password"
+        required
+      />
+      <button
+        type="submit"
+        className="w-full p-4 bg-blue-500 text-white rounded-md hover:bg-blue-600 transition"
+      >
+        Zarejestruj się
+      </button>
+      <p className="text-center mt-4">
+        Posiadasz już konto?{' '}
+        <a href="#" onClick={toggleForm} className="text-blue-500 underline">
+          Zaloguj się.
+        </a>
+      </p>
+    </form>
+  )
```
Also this same way as login file, then just import both of them here

KacperWojdak replied:
✔️

KacperWojdak committed:
Seperate Login and Register

Seperate Login Form and Register Form into Components folder and apply them as imports

KacperWojdak committed:
Eslint cleaning

PiotrFedak commented on frontend/src/pages/LoginPage.js:
Can we add some image in the background ? you could use your ai magic tool. 
Or use https://undraw.co/  on this website it is possible to chose what color of svg you want for example your favorite yellow #C58A09

KacperWojdak replied:
✔️

PiotrFedak reviewed (changes requested):
I still don't like the overall looks.
For me the color just don't match with each other, you could create a palet of 3-4 color that will match with your yellow, i guess it is something that can be added anytime so you can take your time if you have to.

![a13a0132ead49fb38fcce528c7c75521](https://github.com/user-attachments/assets/8f2b766b-8379-4e03-8bed-9cc392d4978b)

This is the contrast score of you header

KacperWojdak committed:
Changes

Changed colours pallete, improved an auth form with images and animations

KacperWojdak commented:
![1](https://github.com/user-attachments/assets/1cade256-13ef-433d-a385-82f8391b3006)
![2](https://github.com/user-attachments/assets/d9f3bdda-3e2a-481e-9d9a-e73eb784e2e0)

PiotrFedak commented on frontend/src/pages/AuthPage.js:
![79838cca41904682dfa2c056e09c85d8](https://github.com/user-attachments/assets/2087a0bd-a6cb-4785-8d8b-d7c4bcabe9cc)
It is possible to scroll into nothing, you need to add `overflow-hidden` to block that possibility. Or create a footer that will take the bottom space

KacperWojdak replied:
✔️

KacperWojdak committed:
Added Footer

KacperWojdak committed:
Added footer into Home and Map page + eslint

TomaszBarczycki reviewed (commented):
Apart from everything our PM mentioned, everything seems fine to me

PiotrFedak reviewed (changes requested):
There is a lack of axios that is needed for register and login to work.
I would suggest to create a axiosClient.js the way we do in other projects.
Also add "ContextProvider.js" maybe even create a GuestLayout to try if blocking path works.

Of course if you don't like that approach you could try with Auth0 or some other authentication service.

KacperWojdak committed:
Axios login register setup

KacperWojdak committed:
Working Axios

Register and login working correctly, after login redirect to /map, changed button on Home from "Logowanie" to "Wyloguj" as user is logged in. Also blocked possibility to write /auth into search bar as a logged in user

KacperWojdak committed:
eslint

TomaszBarczycki reviewed (changes requested):
I'd consider to make the app responsive. When you try to open it with a mobile view, there are a lot of bugs

KacperWojdak committed:
ContextProvider

KacperWojdak committed:
Eslint fixes

KacperWojdak commented:
![resp1](https://github.com/user-attachments/assets/f60b6257-7349-44cd-888b-3da425594960)
![resp2](https://github.com/user-attachments/assets/2571ba11-d5ef-4475-b5e8-62bc0468a391)
![resp3](https://github.com/user-attachments/assets/fe7c279d-1940-4676-b3cc-847d59d79f57)
![resp4](https://github.com/user-attachments/assets/7254e541-b115-43dc-b5fd-2ad1867774b5)

For responsive mobile view. Header is changing button text into powrót and making title smaller. On main page a panel is getting narrower and buttons are one by one. From AuthPage images are gone so there is only a login/register form and on map page there is a map and list one by one for now.

KacperWojdak committed:
Small cleanup / font size on mainpage

KacperWojdak committed:
cleanup

PiotrFedak commented on frontend/src/index.js:
```diff
@@ -2,15 +2,18 @@ import React from 'react'
 import ReactDOM from 'react-dom/client'
 import './index.css'
 import App from './App.js'
+import { ContextProvider } from './ContextProvider.js'
 
 const root = ReactDOM.createRoot(document.getElementById('root'))
 root.render(
   <React.StrictMode>
-    <App />
+    <ContextProvider>
+      <App />
+    </ContextProvider>
   </React.StrictMode>,
 )
 
 // If you want to start measuring performance in your app, pass a function
 // to log results (for example: reportWebVitals(console.log))
 // or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
-reportWebVitals()
+//reportWebVitals()
```
Import router here in this file instead of App

KacperWojdak replied:
✔️

PiotrFedak commented on frontend/src/App.test.js:
```diff
@@ -1,4 +1,5 @@
 import { render, screen } from '@testing-library/react'
+import { expect, test } from '@jest/globals';
 import App from './App'
 
 test('renders learn react link', () => {
```
Also this, can be remove

KacperWojdak replied:
✔️

PiotrFedak commented on frontend/src/App.js:
```diff
@@ -1,5 +1,5 @@
 import React from 'react'
-import AppRouter from './router'
+import AppRouter from './router.js'
 
 export default function App() {
   return (
```
When you done moving routing to index. then this file can be deleted

KacperWojdak replied:
✔️

PiotrFedak reviewed (commented):
If u gonna repat some colors like "#DDA15E" then you could add this to tailwind css config file and name it something norumal.

KacperWojdak committed:
Made custom colors into tailwindconfig.  Deleted App.js and Apptest so routing is directly in index.js
